src/TBuffer.cpp

```diff
@@ -30,6 +29,552 @@
 
 #include <assert.h>
 
+// Define this to get qDebug() messages about the decoding of UTF-8 data when it
+// is not the single bytes of pure ASCII text:
+#define DEBUG_UTF8_PROCESSING
```

Comment:
What is this trying to say? When UTF-8 data is more complicated than just ASCII?